src/mudlet.cpp

```diff
@@ -4322,6 +4322,161 @@ void mudlet::doAutoLogin(const QString& profile_name)
     enableToolbarButtons();
 }
 
+// Telnet URI handling implementation
+// Parse a telnet:// URI according to RFC 4248
+mudlet::TelnetUriData mudlet::parseTelnetUri(const QString& uri)
+{
+    TelnetUriData data;
+    
+    QUrl url(uri);
+    
+    // Validate scheme
+    if (url.scheme().toLower() != qsl("telnet")) {
+        qDebug() << "mudlet::parseTelnetUri() - Invalid URI scheme:" << url.scheme() << "expected 'telnet'";
+        data.valid = false;
+        return data;
+    }
+    
+    // Extract host
+    data.host = url.host();
+    if (data.host.isEmpty()) {
+        qDebug() << "mudlet::parseTelnetUri() - URI missing host";
+        data.valid = false;
+        return data;
+    }
+    
+    // Extract port (default to 23 per RFC 4248)
+    data.port = url.port(23);
+    
+    // Extract optional username and password (rarely used per RFC 4248)
+    if (!url.userName().isEmpty()) {
+        data.username = url.userName();
+    }
+    if (!url.password().isEmpty()) {
+        data.password = url.password();
+    }
+    
+    data.valid = true;
+    qDebug() << "mudlet::parseTelnetUri() - Parsed URI successfully: host=" << data.host << "port=" << data.port;
+    return data;
+}
+
+// Find existing profile matching host and port
+QString mudlet::findMatchingProfile(const QString& host, int port)
+{
+    QDir profilesDir(getMudletPath(enums::profilesPath));
+    QStringList profileNames = profilesDir.entryList(QDir::Dirs | QDir::NoDotAndDotDot, QDir::Name);
+    
+    // Also check predefined game profiles
+    profileNames += TGameDetails::keys();
+    profileNames.removeDuplicates();
+    
+    QString matchedProfile;
+    QDateTime latestTime;
+    
+    for (const auto& profileName : profileNames) {
+        QString profileHost = readProfileData(profileName, qsl("host_url"));
+        QString profilePort = readProfileData(profileName, qsl("port_number"));
+        
+        // Match based on host and port
+        if (profileHost.toLower() == host.toLower() && profilePort.toInt() == port) {
```

Comment:
That'll hit a Clazy warning about unneeded case conversions. :grin::
```suggestion
        if (!profileHost.compare(host, Qt::CaseInsensitive) && profilePort.toInt() == port) {
```